community/cypher/runtime-util/src/main/java/org/neo4j/cypher/internal/runtime/LongArrayHashTable.java

```diff
@@ -0,0 +1,174 @@
+     * addition to a hash table also a separate array for the values.
+     *
+     * @return true if the number of keys in the hash table has reached capacity.
+     */
+    boolean timeToResize()
+    {
+        return numberOfEntries == resizeLimit;
+    }
+
+    /***
+     * Checks whether a slot in the table contains a given key.
+     * @param slot Slot to check
+     * @param key Key to check
+     * @return Can return:
+     *  SLOT_EMPTY - This slot is free. In other words - the key is not in the table
+     *  CONTINUE_PROBING - This slot is taken by a different key
+     *  VALUE_FOUND - The key is in the table at this slot
+     */
+    int checkSlot( int slot, long[] key )
+    {
+        assert LongArrayHash.validValue( key, width );
+
+        int startOffset = slot * width;
+        if ( keys[startOffset] == NOT_IN_USE )
+        {
+            return SLOT_EMPTY;
+        }
+
+        for ( int i = 0; i < width; i++ )
+        {
+            if ( keys[startOffset + i] != key[i] )
+            {
+                return CONTINUE_PROBING;
+            }
+        }
+
+        return VALUE_FOUND;
+    }
+
+    /**
+     * Writes the key to this slot.
+     *
+     * @param slot The slot to write to.
+     * @param key Le key.
+     */
+    void claimSlot( int slot, long[] key )
+    {
+        int offset = slot * width;
+        System.arraycopy( key, 0, keys, offset, width );
+        numberOfEntries++;
+    }
+
+    public boolean isEmpty()
+    {
+        for ( int i = 0; i < keys.length; i = i + width )
+        {
+            if ( keys[i] != NOT_IN_USE )
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Finds an slot not already claimed, starting from a given slot.
+     *
+     * @return First unused slot after fromSlot
+     */
+    private int findUnusedSlot( int fromSlot )
+    {
+        while ( true )
+        {
+            if ( keys[fromSlot * width] == NOT_IN_USE )
+            {
+                return fromSlot;
+            }
+            fromSlot = (fromSlot + 1) & tableMask;
+        }
+    }
+
+    LongArrayHashTable doubleCapacity()
+    {
+        LongArrayHashTable newTable = new LongArrayHashTable( capacity * 2, width );
+        newTable.numberOfEntries = numberOfEntries;
+
+        for ( int fromOffset = 0; fromOffset < capacity * width; fromOffset = fromOffset + width )
+        {
+            if ( keys[fromOffset] != NOT_IN_USE )
+            {
+                int toSlot = LongArrayHash.hashCode( keys, fromOffset, width ) & newTable.tableMask;
+                toSlot = newTable.findUnusedSlot( toSlot );
+                System.arraycopy( keys, fromOffset, newTable.keys, toSlot * width, width );
+            }
+        }
+
+        return newTable;
+    }
+
+    Pair<LongArrayHashTable,Object[]> doubleCapacity( Object[] srcValues )
```

Comment:
In this method, changing `src*` to `from*` and `dst*` to `to*` would give better consistency.